src/rospack.cpp

```diff
@@ -2060,8 +2060,8 @@ Rosstackage::writeCache()
       FILE *cache = fopen(tmp_cache_path, "w");
 #else
     mode_t mask = umask(S_IRWXU);
-    int fd = mkstemp(tmp_cache_path);
     umask(mask);
+    int fd = mkstemp(tmp_cache_path);
```

Comment:
This change doesn't make sense. Please read the documentation of `umask`: http://man7.org/linux/man-pages/man2/umask.2.html

The first cal in line 2062 actually changes the mark to `S_IRWXU`. The returned value contains the *previous* value of the mask. That previous value is then restored *after* `mkstemp` has been called.

Moving the `mkstemp` *below* the second `umask` calls makes therefore no sense. The two `umask` calls together are basically a no-op.